IDL/qsfit/qsfit.pro

```diff
@@ -232,7 +232,23 @@ FUNCTION qsfit_input, x, y, e, OPT=opt, TYPE=type, ID=id, Z=z, EBV=ebv
   COMMON COM_QSFITOPT, qsfitOpt
 
   IF (gn(opt) EQ 0) THEN qsfitOpt = qsfit_options() ELSE qsfitOpt = opt
-  IF (gn(type) EQ 0) THEN type = 'DATA'
+  IF (gn(type) EQ 0) THEN type = 'DATA' $
+
+  ELSE BEGIN
+
+  ;;Manage tsv and csv ASCII files; json and VOTable are TBD
+     ascii_type = STRUPCASE(type)
+     CASE (ascii_type) OF
+        'TSV': sep = STRING(9b)
+        'CSV': sep = ','
+        'TXT': sep = ' '
+        'ASCII': sep = ' '
+        ELSE: sep = ''
+     ENDCASE
+
+     IF (sep NE '') THEN type = 'ASCII'
```

Comment:
This line may affect an external variable used as input to the `TYPE=` keyword.
I suggest to modify the `CASE` statement below into a `SWITCH` statement and introduce the `'TSV'`, `'CSV'` and '`TXT'` cases.  E.g.:
```
SWITH (type) OF
      'DATA': BEGIN
           ...
           BREAK
      END
      'SDSS_DR10': BEGIN
          ....
          BREAK
       END
      'TSV':
      'CSV':
      'TXT':
      'ASCII': BEGIN
             .....
       END
ENDSWITCH
```

Reply:
Switch to SWITCH

For additional file reading see Issues.